ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/OffsetBuilderSpec.scala

```diff
@@ -14,67 +14,68 @@ class OffsetBuilderSpec extends AnyWordSpec with Matchers with ScalaCheckDrivenP
 
   "OffsetBuilder" should {
     "return all zeroes for the zeroth offset" in {
-      val offset = OffsetBuilder.fromLong(0)
+      val offset = OffsetBuilder.of(OffsetBuilder.DefaultVersion, 0)
 
       offset should be(zeroOffset)
     }
 
     "always return an offset of the same length" in {
-      forAll { (highest: Long, middle: Int, lowest: Int) =>
-        val offset = OffsetBuilder.fromLong(highest, middle, lowest)
+      forAll { (version: Byte, highest: Int, middle: Int, lowest: Int) =>
```

Comment:
I changed the `highest` to `Int` to reduce the boilerplate. If you're not fine with that, I'll have to use `whenever` or something similar.

Reply:
Please use `whenever`, or preferably, `Gen.chooseNum`. We want to make sure that everything in the range works.

Something like this should work.

```suggestion
      val genHighest = Gen.chooseNum(0L, (1L << 56) - 1)
      forAll(arbitrary[Byte], genHighest, arbitrary[Int], arbitrary[Int]) {
        (version, highest, middle, lowest) =>
```

You can reuse that generator in other tests.